packages/x-date-pickers/src/internals/hooks/usePicker/usePickerViews.ts

```diff
@@ -59,6 +60,11 @@ export interface UsePickerViewsBaseProps<
    * If `undefined`, internally defined view will be the used.
    */
   viewRenderers: PickerViewRendererLookup<TValue, TView, TExternalProps, TAdditionalProps>;
+  /**
+   * If `true`, the toolbar will be visible.
+   * @default `true` for mobile, `false` for desktop
+   */
+  showToolbar?: boolean;
```

Comment:
Heads-up: This is either going to conflict with https://github.com/mui/mui-x/pull/7498

Reply:
An initial idea was to show view switcher only if:
- There is view to swich (more than 1 time view)
- The swich is not already available in the toolbar

Since `showToolbar` is not available any more, and using `slotProps.toolbar.hidden` does not seem reliable (as a developer it would be strange that a slot props impact another one) I propose to go to the safest way: show the view swich even if there is a toolbar. Leading to the following argos diff

@LukasTy does it seems good for you?

![image](https://user-images.githubusercontent.com/45398769/212922446-23001943-03ef-496e-8d5f-83ffbee8e16a.png)

Reply:
> Since `showToolbar` is not available any more, and using `slotProps.toolbar.hidden` does not seem reliable (as a developer it would be strange that a slot props impact another one)

Hm. Why do you think like that? `showToolbar` was being used to understand if this view switcher need to be shown, but the only thing that changed is the location this prop has moved to.
If we want/need the view switcher logic to depend on toolbar prop, I do not see much problem with it. 🤔 
Maybe adding a mention about this relationship somewhere would be useful, but I don't know the best place for it. 🙈

Reply:
Note that this kind of scenario won't work well if we support the callback format at somepoint.
Because we would not have the `ownerState` outside of the layout component, so we could not correctly resolve the `toolbar.hidden` prop.

Reply:
Yes, I did not thaught about that edge case